core/imodel-bridge/src/BADGERSIssueReporter.ts

```diff
@@ -0,0 +1,288 @@
+            "file": {
+                "identifier": sourceFile.fileId,
+                "path": sourceFile.path,
+                "ignoredelements": ignoredelements,
+            },
+            "auditrecords": auditrecords
+        }];
+
+        const report = {
+            "context": context,
+            "sourceFilesInfo": sourceFile,
+            "filerecords": fileRecord
+        }
+        const reportFilePath = path.join(this._stagingDir, this._reportFile);
+        fs.writeFileSync(reportFilePath, JSON.stringify(report));
+        return report;
+    }
+
+    private issueExistsInDb(ecInstanceId: string, sourceId: string, level: string, category: string, message: string, type: string) {
+        let exists = false;
+        const values = [this._fileInfo.fileId, ecInstanceId, sourceId, level, category, message, type];
+        this._reportDb.withPreparedSqliteStatement(`SELECT * FROM ${this.TEMP_REPORT_AUDIT_RECORDS_TABLE} WHERE fileId =? and ecInstanceId =? and elementSourceId =? and auditLevel =? and auditCategory =? and auditMessage =? and auditType =? `, (stmt) => {
+            stmt.bindValues(values);
+            if (stmt.step() === DbResult.BE_SQLITE_ROW)
+                exists = true;
+        })
+        return exists;
+    }
+
+    private handleFileRecord() {
+        this._reportDb.withPreparedSqliteStatement(`SELECT fileId FROM ${this.TEMP_REPORT_FILE_RECORDS_TABLE} WHERE fileId =? and activityId =? `, (stmt) => {
+            stmt.bindValue(1, this._fileInfo.fileId);
+            stmt.bindValue(2, this._contextInfo.activityId);
+            if (stmt.step() !== DbResult.BE_SQLITE_ROW) {
+                this._reportDb.withSqliteStatement(`INSERT INTO ${this.TEMP_REPORT_FILE_RECORDS_TABLE}(activityId, fileId, filePath) VALUES(?,?,?)`, (myStmt) => {
+                    const values = [this._contextInfo.activityId, this._fileInfo.fileId, this._fileInfo.filePath]
+                    myStmt.bindValues(values);
+                    myStmt.step();
+                })
+            }
+        })
+    }
+
+    private createDBSchema() {
+        this._reportDb.withSqliteStatement(`CREATE TABLE ${this.TEMP_REPORT_FILE_RECORDS_TABLE}(activityId STRING, fileId STRING NOT NULL UNIQUE, filePath STRING)`, (stmt) => { stmt.step(); });
+        this._reportDb.withSqliteStatement(`CREATE INDEX ${this.TEMP_REPORT_FILE_RECORDS_TABLE}ActivityIdx ON ${this.TEMP_REPORT_FILE_RECORDS_TABLE}(activityId)`, (stmt) => { stmt.step(); });
+
+        this._reportDb.withSqliteStatement(`CREATE TABLE ${this.TEMP_REPORT_AUDIT_RECORDS_TABLE}(fileId STRING NOT NULL, ecInstanceId STRING, elementSourceId STRING, auditLevel STRING,
+            auditCategory STRING, auditMessage STRING, auditType STRING, GUID STRING)`, (stmt) => { stmt.step(); });
+
+        this._reportDb.withSqliteStatement(`CREATE TABLE ${this.SRCFILE_TEMP_BRIDGE_ISSUES_TABLE}(activityId STRING, contextId STRING, jobId STRING, imodelId STRING, briefcaseId STRING,
```

Comment:
I'd consider having a unique constraint on the various fields in the table so you don't have to manually do the SELECT check. A quick google search for syntax: https://www.sqlitetutorial.net/sqlite-unique-constraint/